src/framework/components/anim/component-binder.js

```diff
@@ -67,36 +67,51 @@ class AnimComponentBinder extends DefaultAnimBinder {
     }
 
     resolve(path) {
-        var propertyLocation = AnimBinder.decode(path);
-
-        var entity = this._getEntityFromHierarchy(propertyLocation.entityPath);
-
-        if (!entity)
-            return null;
+        var encodedPath = AnimBinder.encode(path.entityPath, path.component, path.propertyPath);
+        var target = this.targetCache[encodedPath];
+        if (target) return target;
 
+        var entity;
         var propertyComponent;
+        var targetPath;
 
-        switch (propertyLocation.component) {
+        switch (path.component) {
             case 'entity':
+                entity = this._getEntityFromHierarchy(path.entityPath);
+                targetPath = AnimBinder.encode(
+                    entity.path,
+                    'entity',
+                    path.propertyPath
+                );
                 propertyComponent = entity;
                 break;
             case 'graph':
-                if (entity.model && entity.model.model && entity.model.model.graph) {
-                    propertyComponent = entity.model.model.graph.findByPath(propertyLocation.entityPath[0]);
-                }
-                if (!propertyComponent) {
-                    var entityPath = AnimBinder.splitPath(propertyLocation.entityPath[0], '/');
-                    propertyComponent = this.nodes[entityPath[entityPath.length - 1] || ""];
-                }
+                entity = this.animComponent.entity;
+                var graph = entity.model && entity.model.model && entity.model.model.graph;
```

Comment:
You could potentially use optional chaining here:

```javascript
                var graph = entity.model?.model?.graph;
```

Reply:
I see this handles entity.model ... and I hope there's handling of entity.render as well for the new render component?

Reply:
I've made this more generic so it'll use the entity as the base of the graph when no model component is present, tested and working 👍